### PR TITLE
Add in memory hash table to hold last received temperature data

### PR DESCRIPTION
**Pull request by charithag**

### CODE REVIEW

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 1]
```diff
@@ -505,7 +504,9 @@ public void switchBulb(@HeaderParam("owner") String owner,
 		String protocolString = protocol.toUpperCase();
 		String callUrlPattern = VirtualFireAlarmConstants.BULB_CONTEXT + switchToState;
 
-		log.info("Sending request to switch-bulb of device [" + deviceId + "] via " + protocolString);
+        if (log.isDebugEnabled()) {
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:507]: fix formatting issues on indentation.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 2]
```diff
@@ -209,8 +208,8 @@ public void removeDevice(@PathParam("device_id") String deviceId,
 		deviceIdentifier.setType(VirtualFireAlarmConstants.DEVICE_TYPE);
 		try {
 			boolean removed = deviceManagement.getDeviceManagementService().disenrollDevice(
-					deviceIdentifier);
-			if (removed) {
+                    deviceIdentifier);
+            if (removed) {
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:212]: fix formatting issues on indentation.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 3]
```diff
@@ -559,9 +560,12 @@ public String requestSonarReading(@HeaderParam("owner") String owner,
 		}
 
 		String protocolString = protocol.toUpperCase();
-		log.info("Sending request to read sonar value of device [" + deviceId + "] via " + protocolString);
 
-		try {
+        if (log.isDebugEnabled()) {
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:564]: fix formatting issues on indentation.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 4]
```diff
@@ -559,9 +560,12 @@ public String requestSonarReading(@HeaderParam("owner") String owner,
...
+            log.debug("Sending request to read sonar value of device [" + deviceId + "] via " + protocolString);
+        }
+
+        try {
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:568]: fix formatting issues on indentation.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 5]
```diff
@@ -601,40 +605,40 @@ public String requestSonarReading(@HeaderParam("owner") String owner,
...
 	@GET
-	public String requestTemperature(@HeaderParam("owner") String owner,
+    @Consumes("application/json")
+    @Produces("application/json")
+	public TemperatureRecord requestTemperature(@HeaderParam("owner") String owner,
 	                                 @HeaderParam("deviceId") String deviceId,
 	                                 @HeaderParam("protocol") String protocol,
 	                                 @Context HttpServletResponse response) {
-		String replyMsg = "";
+        TemperatureRecord temperatureRecord = null;
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:614]: fix formatting issues on indentation.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 6]
```diff
@@ -601,40 +605,40 @@ public String requestSonarReading(@HeaderParam("owner") String owner,
...
 			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
-			return replyMsg;
 		}
 
 		String protocolString = protocol.toUpperCase();
 
-		log.info("Sending request to read virtual-firealarm-temperature of device [" + deviceId + "] via " + protocolString);
+        if (log.isDebugEnabled()) {
+            log.debug("Sending request to read virtual-firealarm-temperature of device [" + deviceId + "] via " + protocolString);
+        }
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:629]: fix formatting issues on indentation.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 7]
```diff
@@ -601,40 +605,40 @@ public String requestSonarReading(@HeaderParam("owner") String owner,
...
 	                                 @HeaderParam("deviceId") String deviceId,
 	                                 @HeaderParam("protocol") String protocol,
 	                                 @Context HttpServletResponse response) {
-		String replyMsg = "";
+        TemperatureRecord temperatureRecord = null;
 
 		DeviceValidator deviceValidator = new DeviceValidator();
 		try {
 			if (!deviceValidator.isExist(owner, SUPER_TENANT, new DeviceIdentifier(deviceId, VirtualFireAlarmConstants.DEVICE_TYPE))) {
 				response.setStatus(Response.Status.UNAUTHORIZED.getStatusCode());
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:619]: put appropriate error log messages as a way of conveying an error has occurred at server-side.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 8]
```diff
@@ -601,40 +605,40 @@ public String requestSonarReading(@HeaderParam("owner") String owner,
...
-				return "Unauthorized Access";
 			}
 		} catch (DeviceManagementException e) {
-			replyMsg = e.getErrorMessage();
 			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:622]: put appropriate error log messages as a way of conveying an error has occurred at server-side.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 9]
```diff
@@ -601,40 +605,40 @@ public String requestSonarReading(@HeaderParam("owner") String owner,
...
+            log.debug("Sending request to read virtual-firealarm-temperature of device [" + deviceId + "] via " + protocolString);
+        }
 
 		try {
 			switch (protocolString) {
 				case HTTP_PROTOCOL:
 					String deviceHTTPEndpoint = deviceToIpMap.get(deviceId);
 					if (deviceHTTPEndpoint == null) {
-						replyMsg = "IP not registered for device: " + deviceId + " of owner: " + owner;
 						response.setStatus(Response.Status.PRECONDITION_FAILED.getStatusCode());
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:636]: put appropriate error log messages as a way of conveying an error has occurred at server-side.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 10]
```diff
@@ -601,40 +605,40 @@ public String requestSonarReading(@HeaderParam("owner") String owner,
...
-						return replyMsg;
-					}
-
-					replyMsg = sendCommandViaHTTP(deviceHTTPEndpoint, VirtualFireAlarmConstants.TEMPERATURE_CONTEXT, false);
-					break;
+                    }
+					String tString = sendCommandViaHTTP(deviceHTTPEndpoint, VirtualFireAlarmConstants.TEMPERATURE_CONTEXT, false);
+                    float temperature = Float.parseFloat(tString);
+                    DataHolder.getThisInstance().setTemperature(deviceId, temperature, Calendar.getInstance().getTimeInMillis());
+                    break;
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:641]: fix formatting issues on indentation.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 11]
```diff
@@ -645,20 +649,16 @@ public String requestTemperature(@HeaderParam("owner") String owner,
 					break;
 
 				default:
-					replyMsg = "Requested protocol '" + protocolString + "' is not supported";
-					response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
-					return replyMsg;
-			}
+                    response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:652]: put appropriate error log messages as a way of conveying an error has occurred at server-side.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 12]
```diff
@@ -645,20 +649,16 @@ public String requestTemperature(@HeaderParam("owner") String owner,
...
+            }
+            temperatureRecord = DataHolder.getThisInstance().getTemperature(deviceId);
 		} catch (DeviceManagementException e) {
-			replyMsg = e.getErrorMessage();
 			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:656]: put appropriate error log messages as a way of conveying an error has occurred at server-side to system administrators.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java [hunk 13]
```diff
@@ -684,7 +684,7 @@ public void pushTemperatureData(final DeviceJSON dataMsg, @Context HttpServletRe
 			response.setStatus(Response.Status.CONFLICT.getStatusCode());
 			return;
 		}
-
+        DataHolder.getThisInstance().setTemperature(deviceId, temperature, Calendar.getInstance().getTimeInMillis());
 		if (!publishToDAS(dataMsg.owner, dataMsg.deviceId, dataMsg.value)) {
 			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:689]: put appropriate error log messages as a way of conveying an error has occurred at server-side to system administrators.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/DataHolder.java [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+public class DataHolder {
+
+    private static DataHolder thisInstance = new DataHolder();
+    private Map<String, TemperatureRecord> temperatureMap = new HashMap<>();
+
+    private DataHolder() {
+
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/DataHolder.java:30]: unnecessary new line.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/VirtualFireAlarmMQTTSubscriber.java [hunk 15]
```diff
@@ -68,10 +64,14 @@ protected void postMessageArrived(String topic, MqttMessage message) {
 			if(log.isDebugEnabled()) {
 				log.debug("MQTT Subscriber: Published data to DAS successfully.");
 			}
-		} else {
-			log.info("MQTT: Reply Message [" + message.toString() + "] topic: [" + topic + "]");
-		}
-	}
+        } else if (message.toString().contains("TEMPERATURE")) {
+            log.info("MQTT: Reply Message [" + message.toString() + "] topic: [" + topic + "]");
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/VirtualFireAlarmMQTTSubscriber.java:68]: avoid unnecessary info messages as much as possible. These type of regular messages would be mostly appropriate in a debugging context.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/VirtualFireAlarmMQTTSubscriber.java [hunk 16]
```diff
@@ -68,10 +64,14 @@ protected void postMessageArrived(String topic, MqttMessage message) {
...
+            float temperature = Float.parseFloat(message.toString().split(":")[1]);
+            DataHolder.getThisInstance().setTemperature(deviceId, temperature, Calendar.getInstance().getTimeInMillis());
+        } else {
+            log.info("MQTT: Message [" + message.toString() + "] topic: [" + topic + "]");
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/VirtualFireAlarmMQTTSubscriber.java:72]: avoid unnecessary info messages as much as possible. These type of regular messages would be mostly appropriate in a debugging context.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 2 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:212]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 1 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:507]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 3 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:564]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 4 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:568]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 5 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:614]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 7 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:619]: AFAIK, sending unauthorized status to client side should not be an error which consider as an error in server side. Hence not needed to log as an error in server side.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 8 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:622]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 6 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:629]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 9 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:636]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 10 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:641]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 11 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:652]: AFAIK, sending not acceptable status to client side should not be an error which consider as an error in server side. Hence not needed to log as an error in server side.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 12 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:656]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java (hunk 13 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmService.java:689]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/DataHolder.java (hunk 14 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/DataHolder.java:30]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/VirtualFireAlarmMQTTSubscriber.java (hunk 15 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/VirtualFireAlarmMQTTSubscriber.java:68]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/VirtualFireAlarmMQTTSubscriber.java (hunk 16 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/util/VirtualFireAlarmMQTTSubscriber.java:72]: Can't fix as this code is obsolete and no longer exists.

**Pull request merged**